src/main/java/cs361/battleships/models/Board.java

```diff
@@ -27,16 +28,23 @@ public Board() {
 	DO NOT change the signature of this method. It is used by the grading scripts.
 	 */
 	public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
-		if (ships.size() >= 3) {
+		if (ships.size() >= 4) {
 			return false;
 		}
 		if (ships.stream().anyMatch(s -> s.getKind().equals(ship.getKind()))) {
 			return false;
 		}
 		final var placedShip = new Ship(ship.getKind());
+		if(ship.isUnderwater()){
+			placedShip.goUnderwater();
+		}
 		placedShip.place(y, x, isVertical);
-		if (ships.stream().anyMatch(s -> s.overlaps(placedShip))) {
-			return false;
+		for(int i=0;i<ships.size();i++){// Rewritten to check underwater status.
+			if(ships.get(i).overlaps(placedShip)){// If there is an overlap
+				if(ships.get(i).isUnderwater() == placedShip.isUnderwater()) {// If they are both under/above water error out
+					return false;
+				}//Otherwise we are good
+			}
```

Comment:
Good check to make sure two ships in the same position on board and same level (submerged/surface) can't be placed.